File: src/nerd/tuxmobil/fahrplan/camp11/Lecture.java
package nerd.tuxmobil.fahrplan.camp11;

import android.text.format.Time;

public class Lecture {
	public String title;
	public String subtitle;
	public int day;
	public String room;
	public int startTime;		// minutes since day start
	public int duration;		// minutes
	public String speakers;
	public String track;
	public String lecture_id;
	public String type;
	public String lang;
	public String abstractt;
	public String description;
	public int relStartTime;
	public String links;
	public String date;
	public boolean highlight;
	public boolean has_alarm;
	
	public Lecture(String lecture_id) {
		title = "";
		subtitle = "";
		day = 0;
		room = "";
		startTime = 0;
		duration = 0;
		speakers = "";
		track = "";
		type = "";
		lang = "";
		abstractt = "";
		description = "";
		relStartTime = 0;
		links = "";
		date = "";
		this.lecture_id = lecture_id;
		highlight = false;
		has_alarm = false;
	}

	public static int parseStartTime(String text) {
		String time[] = text.split(":");
		return Integer.parseInt(time[0])*60 + Integer.parseInt(time[1]);
	}

	public static int parseDuration(String text) {
		String time[] = text.split(":");
		return Integer.parseInt(time[0])*60 + Integer.parseInt(time[1]);
	}
	
	public Time getTime() {
		Time t = new Time();
		String[] splitDate = date.split("-");
		t.setToNow();
		t.year = Integer.parseInt(splitDate[0]);
		t.month = Integer.parseInt(splitDate[1])-1;
		t.monthDay = Integer.parseInt(splitDate[2]);
		t.hour = relStartTime / 60;
		t.minute = relStartTime % 60;
		
		return t;
	}
}

File: network/src/main/java/info/metadude/android/eventfahrplan/network/models/Lecture.kt
package info.metadude.android.eventfahrplan.network.models

data class Lecture(

        var eventId: String = "",
        var abstractt: String = "",
        var dayIndex: Int = 0,
        var date: String = "",
        var dateUTC: Long = 0,
        var description: String = "",
        var duration: Int = 0, // minutes
        var hasAlarm: Boolean = false,
        var isHighlight: Boolean = false,
        var language: String = "",
        var links: String = "",
        var relativeStartTime: Int = 0,
        var recordingLicense: String = "",
        var recordingOptOut: Boolean = RECORDING_OPT_OUT_OFF,
        var room: String = "",
        var roomIndex: Int = 0,
        var speakers: String = "",
        var startTime: Int = 0, // minutes since day start
        var slug: String = "",
        var subtitle: String = "",
        var title: String = "",
        var track: String = "",
        var type: String = "",

        var changedDayIndex: Boolean = false,
        var changedDuration: Boolean = false,
        var changedIsCanceled: Boolean = false,
        var changedIsNew: Boolean = false,
        var changedLanguage: Boolean = false,
        var changedRecordingOptOut: Boolean = false,
        var changedRoom: Boolean = false,
        var changedSpeakers: Boolean = false,
        var changedStartTime: Boolean = false,
        var changedSubtitle: Boolean = false,
        var changedTitle: Boolean = false,
        var changedTrack: Boolean = false
) {

    companion object {
        const val RECORDING_OPT_OUT_OFF = false

        fun parseStartTime(text: String): Int {
            val time = text.split(":").toTypedArray()
            return Integer.parseInt(time[0]) * 60 + Integer.parseInt(time[1])
        }

        fun parseDuration(text: String): Int {
            val time = text.split(":").toTypedArray()
            return Integer.parseInt(time[0]) * 60 + Integer.parseInt(time[1])
        }
    }

}
